File: 2144-minimum-cost-of-buying-candies-with-discount/2144-minimum-cost-of-buying-candies-with-discount.cpp
class Solution {
public:
    int minimumCost(vector<int>& cost) {
        sort(cost.begin(), cost.end(), greater());
        int result=0;
        int count=0;
        
        for(int i=0; i<cost.size();i++)
        {
            if(count==2)
            {
                count=0;
                continue;
            }
            result+=cost[i];
            count++;
        }
        return result;
    }
};

File: 2144-minimum-cost-of-buying-candies-with-discount/2144-minimum-cost-of-buying-candies-with-discount.java
class Solution {
    public int minimumCost(int[] cost) {
        Arrays.sort(cost);
        int result=0;
        
        int count=cost.length-1;
        while(count>=0)
        {
            result+=cost[count];
            count--;
            if(count>=0)
            {
                result+=cost[count];
                count-=2;
            }
        }
        return result;
    }
}